pkg/agent/wireguard/client_linux.go

```diff
@@ -145,18 +150,34 @@ func (client *client) Init() error {
 		ListenPort:   &client.wireGuardConfig.Port,
 		ReplacePeers: false,
 	}
-	patch, _ := json.Marshal(map[string]interface{}{
-		"metadata": map[string]interface{}{
-			"annotations": map[string]string{
-				types.NodeWireGuardPublicAnnotationKey: client.privateKey.PublicKey().String(),
+	if !multiclusterEnabled {
+		patch, _ := json.Marshal(map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"annotations": map[string]string{
+					types.NodeWireGuardPublicAnnotationKey: client.privateKey.PublicKey().String(),
+				},
 			},
-		},
-	})
-	if err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		_, err := client.k8sClient.CoreV1().Nodes().Patch(context.TODO(), client.nodeName, apitypes.MergePatchType, patch, metav1.PatchOptions{}, "status")
-		return err
-	}); err != nil {
-		return fmt.Errorf("error when patching the Node with the '%s' annotation: %w", types.NodeWireGuardPublicAnnotationKey, err)
+		})
+		if err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			_, err := client.k8sClient.CoreV1().Nodes().Patch(context.TODO(), client.nodeName, apitypes.MergePatchType, patch, metav1.PatchOptions{}, "status")
+			return err
+		}); err != nil {
+			return fmt.Errorf("error when patching the Node with the '%s' annotation: %w", types.NodeWireGuardPublicAnnotationKey, err)
+		}
+	} else {
+		patch, _ := json.Marshal(map[string]interface{}{
+			"wireGuard": map[string]interface{}{
+				"publicKey": client.privateKey.PublicKey().String(),
```

Comment:
Better to define two constants like `NodeWireGuardPublicAnnotationKey` with domain name. eg: MulticlusterWireGuardPublicAnnotationKey= `multicluster.antrea.io/wireguard-public-key`
MulticlusterWireGuardPortKey = `multicluster.antrea.io/wireguard-port`

Reply:
Add two constant now. It's the Gateway's field, not annotations. So I use "publicKey" and "port" directly.